Mqtt patch

tiagomiotto opened the pull request:
### New framework
Entirely new communication protocol as well as a new modular way to add and interact with other components aside from the IMU, since the objective is to add a proximity sensor and a camera afterward.
The specifics of the implementation can be found in the new readme as well as in the code.

### Edited README
Edited README to be compliant to the changes made to the architecture of the system as well as the implementation.

### Added a rcController
Added a simple keyboard controller to test and demonstrate the communication with the SBrick which controls the motor of the truck.

### Wrote a new IMU script
Wrote a new IMU script to comply with the new architecture, maintaining the old code in the setup function and it is working as before

### Wrote the new main
Wrote a new main that fetches the necessary components from the configuration files, creates the objects and runs them in parallel.
Tested it with the IMU and a dummy component, present in testComponent.py,  and it was working as intended.

### Moved the old code
Moved the legacy code using kafka to the kafka folder, since it can still be used with the frontend repo developed for the sinfo workshop.

tiagomiotto committed:
Updated framework

tiagomiotto committed:
Delete .DS_Store

tiagomiotto committed:
Aded git ignore

tiagomiotto committed:
Merge branch 'mqtt-patch' of https://github.com/tiagomiotto/sinfo-rpi-truck into mqtt-patch

tiagomiotto committed:
Update configuration.py

tiagomiotto commented:
I apologize for the size of the pull request, but since it was a full rebuild of the repo I didn't want to commit it before having it somewhat functional.

necosta reviewed (approved):
I had an overall look and wow, great work! 
LGTM. 

@csazevedo should be able  to make a more detailed review

csazevedo commented on README.md:
```diff
@@ -37,7 +49,88 @@ You should connect:
+### Setting up MQTT Broker
+
+MQTT is a lightweight, low power, easy to implement publish/subscribe protocol which makes it ideal for the M2M(Machine to Machine) communication performed by IOT devices.
+If you are new to MQTT i recommend reading [this article](https://1sheeld.com/mqtt-protocol/).
```
"... to MQTT i..." - capital i

csazevedo commented on controller/controllerMessages.py:
```diff
@@ -0,0 +1,59 @@
+    "sbrick_id": "88:6B:0F:80:29:D1",  # "string. SBrick ID. <sbrick MAC>",
+    "channel": "00",  # " string. hex string. the LEGO power function port you want to drive. <00|01|02|03>",
+    "direction": "00",  # "string. clockwise or counterclockwise. <00|01>",
+    "power": "f6",  # "string. hex string. FF means 100% speed. <00~FF>",
```
100% turn right

csazevedo commented on imuClass.py:
```diff
@@ -0,0 +1,87 @@
+        """
+        self.setup()
+        self.mqttHandler.publish(
+            (self.my_topic+"/pollRate"), self.poll_interval*1.0/1000.0, retain=True)
```
make this a variable so there's no need to repeat it

csazevedo reviewed (approved):
This is a huge improvement to what was already done. I like the structure that you gave to it: looks very intuitive and easy to use.  Good job!

tiagomiotto committed:
Added a variable for the actual polling interval

tiagomiotto committed:
Clearer comments on the controller messages

tiagomiotto committed:
Corrected a typo

tiagomiotto committed:
Calculating the poll interval only once

tiagomiotto committed:
Added clarifying comments to the timestamp

tiagomiotto committed:
Aggregated timestamp patch (#2)

Aligned the sampling ratio of the components in a way that the ratio used for each one is always a multiple of the fastest one, allowing us to join multiple signals based on the same timestamp on the data storage

Also changed the README to reflect the changes in the structure of the components that this caused.

tiagomiotto committed:
Changed Imu sampling rate to 50Hz

tiagomiotto committed:
Fixed main.py
